### PR TITLE
feat: use a gocert client class instead of directly using the HTTP API

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This change introduces a new file similar to the `vault_client` library that abstracts the calls made to the GoCert server

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by saltiyazan — APPROVED** [review]

File: src/client.py
Comment (gruyaume) [inline src/client.py:1]: GoCert would probably a better file name

File: src/client.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def __init__(self, host: str, ca_path: str) -> None:
+        self.host = host
+        self.ca_path = ca_path
+
+    def is_api_available(self) -> bool:
+        """Return if the GoCert server is reachable."""
+        try:
+            req = requests.get(
+                f"https://{self.host}/status",
```
Comment (gruyaume) [inline src/client.py:24]: Here we assume the scheme, I would encourage having this be provided by the user of the class:

```python
client = Gocert(url="https://gocert.com")
```

Also, in the init we can give an example of what we are expecting here in terms of format

File: src/client.py
Comment (gruyaume) [inline src/client.py:1]: This change should impact our unit tests, we shouldn't patch `requests.get` anymore but calls to the GoCert class.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: src/client.py (hunk 1 above)
Comment (kayra1) [reply src/client.py]: done, and added docstring to init

File: src/client.py
Comment (kayra1) [reply src/client.py:1]: done

Comment (kayra1) [reply src/client.py:1]: done

**Review by gruyaume — APPROVED** [review]

**Pull request merged**